### PR TITLE
Generate combined Javadoc documentation

### PR DESCRIPTION
**Pull request by ValentinPostindustria**
[Android issue](https://github.com/prebid/prebid-mobile-android/issues/755). [Original issue](https://github.com/prebid/prebid-mobile-ios/issues/990). 

The URL of the generated documentation: https://docs.prebid.org/prebid-mobile-android/index.html

### CODE REVIEW

File: .github/workflows/upload-docs.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    - name: Commit documentation changes
+      run: |
+        git config --local user.email "41898282+github-actions[bot]@users.noreply.github.com"
+        git config --local user.name "github-actions[bot]"
+        git status
+        TAG_NAME=${{ github.ref }}
+        TAG_NAME=${TAG_NAME#refs/tags/}
+        echo "Current tag: $TAG_NAME"
+        ls -l
+        rm -rf docs
```
Comment (bretg) [inline .github/workflows/upload-docs.yml:39]: Seems like there ought to be a check here to make sure that docs_new actually exists and is of a certain size before removing the current `docs` dir. e.g. use `du -s docs` and abort the operation if less than some reasonable size.

File: .github/workflows/upload-docs.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        cp -r PrebidMobile/build/combinedJavadoc docs_new
+    - name: Checkout repository
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+        ref: docs
+        clean: false
+    - name: Commit documentation changes
+      run: |
+        git config --local user.email "41898282+github-actions[bot]@users.noreply.github.com"
```
Comment (bretg) [inline .github/workflows/upload-docs.yml:32]: Where did this 41898282+github-actions email come from? I guess it's fine, just not familiar with it.

**Review by bretg — CHANGES_REQUESTED** [review]
Amazing progress. I'd imagine there are probably tweaks we'll want to make to the actual docs, but my only substantive comment is to consider making the github action a little more robust.

File: .github/workflows/upload-docs.yml (hunk 2 above)
Comment (ValentinPostindustria) [reply .github/workflows/upload-docs.yml:32]: It's an email for GitHub actions bot. [Ref](https://github.com/marketplace/actions/hexo-action#refer-anchor-1).

**Review by bretg — APPROVED** [review]
Thanks for making the workflow more robust. Note that my review is quite limited. @YuriyVelichkoPI will need to sign off with a more robust review.

**Pull request merged**